### ISSUE TITLE
Equation numbers

### ISSUE THREAD

**Issue by empeirikos** [RickStrahl/MarkdownMonster#803]
I have looked at the documentation for Markdown Monster but I have not found any way to include automatic equation numbering. Have I missed something or is this currently not available?

**Comment by RickStrahl** [comment]
Equation numbering?

Info about using equations are here:

**Comment by empeirikos** [comment]
Rick hi,

Thanks for your email. I have gone through the page you send me. It has a
lot of information about how to write equations, but I cannot see anything
about equation numbers. In standard TeX an equation environment is numbered
automatically (unless the user types equation*) and then a \label{ref} is
used so that the equation number can be accessed using the \ref{ref}
command. I haven’t been able to find anything similar in the documentation
for Markdown Monster. It’s still a powerful program and not having equation
numbers is fairly minor.

Best regards,
Ted

**Comment by RickStrahl** [comment]
We're just exposing what the library ([MathJax](https://www.mathjax.org/)) exposes by default. If you have special needs for your Math rendering you can tweak the configuration. 

What renders in Markdown Monster isn't necessarily what will render when you publish your output since it's just a preview anyway, so it all depends on where you plan on hosting/publishing your Markdown.

IAC, not something that is going to change here.

**Comment by empeirikos** [comment]
Sounds pretty lame but I cannot find the config file for MathJax. Tweaking
it to add numbering is very easy...

Anyways thanks for the suggestion.

**Comment by RickStrahl** [comment]
I'm sorry - I had to check myself how this works, but you can't directly change the way the template is embedded. 

The only way you can do this is by creating a custom theme that always loads the Math library.

* Create a new Preview Theme (by copying an existing one)
* Add the script code into the `Theme.html` template
* Turn of `UseMath` Markdown option
* Switch to the new theme (ie. GithubWithMath or whatever you named it)

The template should be copy of any existing theme and you add the MathJax script into the page:

```html
<!DOCTYPE html>
<!-- saved from url=(0016)http://localhost -->
<html lang="en" xmlns="http://www.w3.org/1999/xhtml">
<head>
  <base href="{$docPath}" />
  <meta charset="utf-8" />

  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <link href="{$themePath}..\scripts\fontawesome\css\font-awesome.min.css" rel="stylesheet" />

  <link href="{$themePath}Theme.css" rel="stylesheet" />

  <script src="{$themePath}..\scripts\jquery.min.js"></script>
  <link href="{$themePath}..\scripts\highlightjs\styles\twilight.css" rel="stylesheet" />
  <script src="{$themePath}..\scripts\highlightjs\highlight.pack.js"></script>
  <script src="{$themePath}..\scripts\preview.js" id="PreviewScript"></script>
  {$extraHeaders}

  <meta http-equiv="CUSTOM" content="CUSTOM" />

  <script type="text/x-mathjax-config">
       // PUT YOUR CONFIG CUSTOMIZATIONS HERE 

      // enable inline parsing with single $ instead of /
      MathJax.Hub.Config({
          tex2jax: {
              // inlineMath: [['$','$'],['\\(','\\)']],
              // displayMath: [ ['$$','$$'], ['\\[','\\]'] ],
              processEscapes: true
          },
      });
      $(document).on('previewUpdated',function() {
          setTimeout(function() {
                      MathJax.Hub.Queue(['Typeset',MathJax.Hub,'#MainContent']);
          },10);
      });
  </script>
  <script async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-MML-AM_CHTML"></script>

</head>
<body>
<div id="MainContent">
    <!-- Markdown Monster Content -->
    {$markdownHtml}
    <!-- End Markdown Monster Content -->
</div>
</body>
</html>
```

The downside of this is that the math library (which is quite large) now loads every time the page is rendered - even if there's no math expression in the page. 

You can localize this to a specific project by creating a `.markdownmonster` file in the root of your math project folder:

```json
{
   "PreviewTheme": "GithubWithMath"
}
```

which isolates the math usage to that project.

**Comment by empeirikos** [comment]
What can I say? Not only you write a great piece of software that people
love to use (my 13 year old son is now writing math on Markdown Monster and
he really enjoys it) but you also spend so much time helping novices like
me. I really appreciate your help.

Best regards,
Ted

**Comment by RickStrahl** [comment]
Hey Ted, thank you...

Let's just say this is not a *novice* type of discussion :smile: This is more of a specialty scenario that unfortunately doesn't have an easier solution than having to create a custom theme.

I might have to think about having a way to provide default templates for a number of integrations. There's the MathJax script as we're doing here, and the Mermaid Charts loader script. Maybe these things should be loaded from a template that can be customized - or use a hardcoded template with an optional override if it exists.
